pkg/buffer/intermediate_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buffer_test
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"testing"
+	"testing/iotest"
+
+	"github.com/palantir/log4j-sniffer/pkg/buffer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// IT WOULD BE GREAT TO SETUP A FUZZING FRAMEWORK WERE WE CAN HAVE A LOAD OF KNOWN INPUT FILES
+// THEN WITH A LOAD OF DIFFERENT BUFFER SIZES, MAKE SURE THEY ARE WORKING WITH THE WRAPPED READER THING
+
+func TestIntermediateBufferReader_ReadingAllContentFromReader(t *testing.T) {
+	t.Run("should error on read error when populating buffer", func(t *testing.T) {
+		expectedErr := errors.New("err")
+		_, err := ioutil.ReadAll(&buffer.IntermediateBufferReader{
+			Reader:      iotest.ErrReader(expectedErr),
+			ContentSize: 1,
+			Buffer:      make([]byte, 5, 5),
+		})
+		assert.Equal(t, expectedErr, err)
+	})
+
+	for _, tc := range []struct {
+		name                   string
+		content                string
+		intermediateBufferSize int
+	}{{
+		name: "empty content, zero sized buffer",
+	}, {
+		name:                   "empty content, non-zero sized buffer buffer",
+		intermediateBufferSize: 7,
+	}, {
+		name:                   "content less than buffer size",
+		content:                "012",
+		intermediateBufferSize: 7,
+	}, {
+		name:                   "content equal to size",
+		content:                "0123456",
+		intermediateBufferSize: 7,
+	}, {
+		name:                   "content larger than buffer size",
```

Comment:
Given how important this is, let's add tests for content +/- 1 in length either side of the buffer size, and the same for exactly double the buffer size

I've read the implementation several times and believe it to be correct in these cases but it would be good to ensure we don't get strange/hard to debug failures around the boundaries.

Reply:
Done.